fortran/src/H5Sff.F90

```diff
@@ -1445,4 +1445,146 @@ END FUNCTION H5Sis_regular_hyperslab
 
   END SUBROUTINE H5Sis_regular_hyperslab_f
 
+!>
+!! \ingroup FH5S
+!!
+!! \brief Closes a dataspace selection iterator.
+!!
+!! \param sel_iter_id Dataspace selection iterator identifier
+!! \param hdferr      \fortran_error
+!!
+!! See C API: @ref H5Ssel_iter_close()
+!!
+  SUBROUTINE h5ssel_iter_close_f(sel_iter_id, hdferr)
+    IMPLICIT NONE
+    INTEGER(HID_T), INTENT(IN) :: sel_iter_id
+    INTEGER, INTENT(OUT) :: hdferr
+    INTERFACE
+       INTEGER(C_INT) FUNCTION H5Ssel_iter_close(sel_iter_id) &
+            BIND(C,NAME='H5Ssel_iter_close')
+         IMPORT :: HID_T, C_INT
+         IMPLICIT NONE
+         INTEGER(HID_T), VALUE :: sel_iter_id
+       END FUNCTION H5Ssel_iter_close
+    END INTERFACE
+
+    hdferr = INT(h5ssel_iter_close(sel_iter_id), C_INT)
+
+  END SUBROUTINE h5ssel_iter_close_f
+
+!>
+!! \ingroup FH5S
+!!
+!! \brief Creates a dataspace selection iterator for a dataspace's selection.
+!!
+!! \param space_id    Dataspace identifier
+!! \param elmt_size   Size of element in the selection
+!! \param flags       Selection iterator flag
+!! \param ds_iter_id  Dataspace selection iterator identifier
+!! \param hdferr      \fortran_error
+!!
+!! See C API: @ref H5Ssel_iter_create()
+!!
+  SUBROUTINE h5ssel_iter_create_f(space_id, elmt_size, flags, ds_iter_id, hdferr)
+    IMPLICIT NONE
+    INTEGER(HID_T) , INTENT(IN)  :: space_id
+    INTEGER(SIZE_T), INTENT(IN)  :: elmt_size
+    INTEGER        , INTENT(IN)  :: flags
```

Comment:
I couldn't find documentation about the values of the flags parameter. C function H5Ssel_iter_create description doesn't talk about the values.

Reply:
I noticed that, too. At the time, I was trying to get AI to return that information for me, but it failed 👎  I was not sure it was a fit for this PR, but I could add the info. Otherwise, I was going to do it in another PR.

Reply:
See https://github.com/HDFGroup/hdf5/blob/develop/src/H5Spublic.h#L55-L70, though this should probably be made more obvious.

Reply:
> I noticed that, too. At the time, I was trying to get AI to return that information for me, but it failed 👎 I was not sure it was a fit for this PR, but I could add the info. Otherwise, I was going to do it in another PR.

Agree, it should be another PR.

Reply:
I think the whole H5S iteration topic is very cryptic and needs to be explained. As a user i would have hard time browsing through the header files :-)

Reply:
I think 0 is also valid; that is one option in the tests. I don't see H5S_SEL_ITER_GET_SEQ_LIST_SORTED being tested, and I've noted the values returned in off and len for * _iter_get* are never checked in the tests.

Reply:
> See https://github.com/HDFGroup/hdf5/blob/develop/src/H5Spublic.h#L55-L70, though this should probably be made more obvious.

Thank you! I didn't know about these flags until this review! Learning HDF5 every day!

Reply:
I updated the docs with valid flags.